editoast/src/views/rolling_stocks/mod.rs

```diff
@@ -167,11 +171,17 @@ async fn update(
 ) -> Result<Json<RollingStockWithLiveries>> {
     let data = data.into_inner();
     let rolling_stock_id = path.into_inner();
-    assert_rolling_stock_unlocked(
-        retrieve_existing_rolling_stock(&db_pool, rolling_stock_id).await?,
-    )?;
+    let existing_rolling_stock =
+        retrieve_existing_rolling_stock(&db_pool, rolling_stock_id).await?;
+    assert_rolling_stock_unlocked(existing_rolling_stock.clone())?;
+    let version = existing_rolling_stock.version;
 
-    let rolling_stock = data.into_rolling_stock_model(rolling_stock_id);
+    let mut rolling_stock = data.into_rolling_stock_model(rolling_stock_id, version);
+    let other_fields_changed = existing_rolling_stock != rolling_stock;
+
+    if other_fields_changed {
```

Comment:
Some nitpicking: this variable is weirdly named, to avoid naming it you could:
```suggestion
    if existing_rolling_stock != rolling_stock {
```